### PR TITLE
Improve CONTRIBUTING docs

### LINKED ISSUE #299: Enhance Community Documents
### Please describe the current behaviour

Dab describes various information with varying levels of detail about how dab works and how to contribute.

### What is suboptimal about the current behaviour

The current behaviour is insufficient because information should be of increasing complexity the further you drill down.

### Describe the way you think it should behave

Enhance community documents (README.md, CONTRIBUTING.md, issue templates, etc.) to provide more information in increasing detail with explanations as to the reasoning behind many of the choices.

### PR DESCRIPTION
**Pull request by Nekroze**
## Change Description

Add section on design and maintainer decisions.

## Relevant Issue(s)

- Closes #299

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -48,14 +48,38 @@ Additionally, to run the suite of regression tests execute:
...
+
+Functionality shared across subcommands should be placed in a library file to be sourced from the [libraries dir](app/lib) however if the functionality is only used by one subcommand then it should exist within the subcommand script itself to keep the definition close to the usage.
+
+## Maintainers
+
+AKA Collaborators in GitHub parlance, these users have the ability to review and accept pull requests to the master and stable branches. This section describes some of the process they will go through when considering a pull request.
+
+Each Pull Request should container the minimum number of commits (each commit being one logical change (meaning no implementing commit plus 2 fix commits, squash them into one) with sufficiently descriptive short (first line of) commit messages suitable to be read on their own in the `dab changelog` or `dab update` output.
```
Comment (Lachlan-S) [inline CONTRIBUTING.md]: This should be contain.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -48,14 +48,38 @@ Additionally, to run the suite of regression tests execute:
...
+
+Completion is handled by "the completion binary" which is a go based application defined in the [completion](completion) directory. The subcommands must be manually defined to match the [subcommands directory tree](app/subcommands), other files in the [completion](completion) directory defined functions or static lists for generating completions on various arguments to subcommands. For example [apps.go](completion/apps.go) defines an array of each app that must be added to when a new app is added to dab, or [repos.go](completion/repos.go) which defines functions to find all defined repos in the users dab config and suggest them for completion.
```
Comment (Lachlan-S) [inline CONTRIBUTING.md]: This should probably be "added to dab. Or [repos.go](completion/repos.go)" since the sentience is just a bit too long.

Comment (Nekroze) [reply CONTRIBUTING.md]: a common issue of mine :smile: thanks!

File: CONTRIBUTING.md (hunk 1 above)
Comment (Nekroze) [reply CONTRIBUTING.md]: durp

**Pull request merged**